packages/runtimes/js/src/JSRuntime.js

```diff
@@ -79,18 +89,29 @@ export default (new Runtime({
 
     let asyncDependencies = [];
     let otherDependencies = [];
-    bundle.traverse(node => {
-      if (node.type !== 'dependency') {
-        return;
-      }
+    let cachedDependencies = bundleDependencies.get(bundle);
 
-      let dependency = node.value;
-      if (dependency.isAsync && !dependency.isURL) {
-        asyncDependencies.push(dependency);
-      } else {
-        otherDependencies.push(dependency);
-      }
-    });
+    if (
```

Comment:
determining these could be moved into a function (e.g. `getDependencies`) and shared with the logic below in `getHintedBundleGroups`